diff --git a/node_modules/data-api-local-serverless/lib/index.js b/node_modules/data-api-local-serverless/lib/index.js
index 0e7f02b..831c1b3 100644
--- a/node_modules/data-api-local-serverless/lib/index.js
+++ b/node_modules/data-api-local-serverless/lib/index.js
@@ -17,6 +17,7 @@ class DataAPILocalServerless {
         const endHandler = this.endHandler.bind(this);
         this.hooks = {
             'before:offline:start:init': startHandler,
+            'before:offline:start': startHandler,
             'before:offline:start:end': endHandler,
             'before:migrations:status:init': startHandler,
             'after:migrations:status:end': endHandler,
